session-helper/flatpak-session-helper.c

```diff
@@ -425,24 +447,88 @@ file_changed (GFileMonitor     *monitor,
               GFile            *file,
               GFile            *other_file,
               GFileMonitorEvent event_type,
-              char             *source)
+              MonitorData      *data)
 {
-  if (event_type == G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT ||
-      event_type == G_FILE_MONITOR_EVENT_CREATED)
-    copy_file (source, monitor_dir);
+  if (event_type != G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT)
+    return;
+
+  char *real = realpath (data->source, NULL);
+  if (real)
+    {
+      /* detect the case that the file's real path has changed and
+         re-add the monitor for the real file */
+      if (g_strcmp0 (data->source, real) &&
+          g_strcmp0 (data->real, real))
+       {
+          free (data->real);
+          data->real = real;
+
+          if (data->monitor_real)
+            {
+              g_signal_handlers_disconnect_by_data (data->monitor_real, data);
+              g_clear_object (&(data->monitor_real));
+            }
+
+          GFile *r = g_file_new_for_path (real);
+          data->monitor_real = g_file_monitor_file (r, G_FILE_MONITOR_NONE, NULL, NULL);
```

Comment:
Nitpick: I’d collect the `GError` here and output it in a `g_debug()` so failures here are easier to debug (and not miss) in future.

Reply:
Was doing what it did before - fixed in both places now. :)